pyro/distributions/transforms/planar.py

```diff
@@ -7,12 +7,67 @@
 from torch.distributions import constraints
 import torch.nn.functional as F
 
+from pyro.distributions.conditional import ConditionalTransform
 from pyro.distributions.torch_transform import TransformModule
 from pyro.distributions.util import copy_docs_from
 
 
 @copy_docs_from(TransformModule)
-class PlanarFlow(TransformModule):
+class ConditionedPlanarFlow(TransformModule):
```

Comment:
I think you'll the `ConditionalPlanarFlow` class to be an `nn.Module` and this class to be a simple `Transform`, right? I guess that means adding a `ConditionalTransformModule` class?

Reply:
I think you're right about this. I'll give it a go making that change. This would mean that `ConditionalTransformedDistribution` would need to be an `nn.Module` as well, just like `TransformedDistribution` is, so you can get the parameters out:

`>>> plf_dist = dist.ConditionalTransformedDistribution(base_dist, [plf])`
`>>> plf_dist.parameters()`

Otherwise there wouldn't be any way to get the parameters out as after making this change the following wouldn't work:

`>>> plf_dist = dist.ConditionalTransformedDistribution(base_dist, [plf]).condition(z)`
`>>> plf_dist.parameters()`

Reply:
Let's separate the two classes, as we do with `Teansform` and `TransformModule`:
```py
class ConditionalTransform(object):
    ...
class ConditionalTransformModule(ConditionalTransform, torch.nn.Module):
```
This avoids forcing a stateful style, and gives user freedom to choose purely functional style.

Reply:
@fritzo I think I've addressed this comment so it's ready for you to look at again! Apologies for the delay